lib/Dialect/FIRRTL/InnerSymbolTable.cpp

```diff
@@ -51,6 +52,25 @@ Operation *InnerSymbolTable::lookup(StringAttr name) const {
   return symbolTable.lookup(name);
 }
 
+/// Get InnerSymbol for an operation.
+StringAttr InnerSymbolTable::getInnerSymbol(Operation *op) {
+  return op->getAttrOfType<StringAttr>(
+      InnerSymbolTable::getInnerSymbolAttrName());
+}
+
+/// Return an InnerRef to the given operation.
+hw::InnerRefAttr InnerSymbolTable::getInnerRef(Operation *op) {
+  assert(op->getParentWithTrait<OpTrait::InnerSymbolTable>() == innerSymTblOp);
+  return hw::InnerRefAttr::get(SymbolTable::getSymbolName(innerSymTblOp),
+                               getInnerSymbol(op));
+}
+hw::InnerRefAttr InnerSymbolTable::getInnerRefFor(Operation *op) {
+  return hw::InnerRefAttr::get(
+      SymbolTable::getSymbolName(
+          op->getParentWithTrait<OpTrait::InnerSymbolTable>()),
+      getInnerSymbol(op));
+}
```

Comment:
Having both methods seems a bit weird, though I grok the utility in have a static method that circumvents the need for constructing the `InnerSymbolTable`.

Reply:
This is weird, I agree.  Pushed a little early to get it out for discussion/sharing after #3406, was still tweaking this particular detail (and thinking how to best support the sort of "get me an InnerRef to this possibly unnamed operation" functionality commonly desired).

How about: drop this variant for now, sinking its implementation into its only user (`InnerSymbol` trait), which is about as convenient (modulo getting it as an `InnerSymbolOpInterface`) anyway.